File: Javascript/String/Delete-Columns-To-Make-Sorted.js
/**
 * @param {string[]} strs
 * @return {number}
 */
const minDeletionSize = (strs) => {
  let count = 0;

  for (index in strs[0]) {
    if (!isSorted(strs.map((s) => s[index]))) {
      count++;
    }
  }
  return count;
};

const isSorted = (str) => {
  for (let i = 0; i < str.length - 1; i++) {
    if (str[i] > str[i + 1]) {
      return false;
    }
  }
  return true;
};

minDeletionSize(["cba", "daf", "ghi"]);

File: Rust/String/Delete-Columns-To-Make-Sorted.rs
pub struct Solution;

impl Solution {
    pub fn min_deletion_size(strs: Vec<String>) -> i32 {
        let mut count = 0;

        for i in 0..strs[0].len() {
            if !strs
                .iter()
                .map(|s| s.as_bytes()[i])
                .collect::<Vec<u8>>()
                .windows(2)
                .all(|pair| pair[0] <= pair[1])
            {
                count += 1
            }
        }

        count
    }
}
fn main() {}
